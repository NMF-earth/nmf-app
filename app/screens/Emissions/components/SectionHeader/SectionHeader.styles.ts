import { StyleSheet } from "react-native";

import { Colors, ComponentsStyle } from "../../../../style";
import { PADDING_HORIZONTAL } from "../../../../constants/Layout";

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    paddingVertical: 10,
    paddingHorizontal: PADDING_HORIZONTAL,
    backgroundColor: Colors.white,
    ...ComponentsStyle.header,
  },
  text: {
    textTransform: "capitalize",
  },
});

export default styles;
