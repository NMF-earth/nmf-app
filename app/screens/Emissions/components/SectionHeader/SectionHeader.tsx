import React from "react";
import { View } from "react-native";
import moment from "moment";

import "moment/min/locales";
import { Text } from "../../../../components";
import styles from "./SectionHeader.styles";
import {
  withLocalization,
  LocalizationContextProps,
  t,
} from "../../../../utils";

interface Props {
  date: Date;
  co2value: number;
  monthlyCarbonBudget: number;
}

const SectionHeader = ({
  date,
  co2value = 1,
  monthlyCarbonBudget = 1,
  language = "",
}: Props & LocalizationContextProps) => {
  let percentageBudget = 1;

  if (monthlyCarbonBudget && co2value) {
    percentageBudget = Math.round((co2value / monthlyCarbonBudget) * 100);
    if (percentageBudget < 1) {
      percentageBudget = 1;
    }
  }

  return (
    <View style={styles.container}>
      <Text.Primary darkGray bold style={styles.text}>
        {moment(date).locale(language).format("MMMM YYYY")}
      </Text.Primary>
      {percentageBudget && (
        <Text.Secondary
          red={percentageBudget > 100}
          green={percentageBudget < 100}
        >
          {percentageBudget + " % " + t("EMISSIONS_SCREEN_HEADER_OF_BUDGET")}
        </Text.Secondary>
      )}
    </View>
  );
};

export default withLocalization(SectionHeader);
