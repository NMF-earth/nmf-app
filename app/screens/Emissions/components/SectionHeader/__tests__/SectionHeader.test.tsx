import React from "react";
import { create } from "react-test-renderer";

import SectionHeader from "../SectionHeader";

jest.unmock("../SectionHeader");

describe("SectionHeader", () => {
  const date = new Date("2020-12-24T03:24:00");

  it("should render when user is over budget", () => {
    const tree = create(
      <SectionHeader co2value={30} monthlyCarbonBudget={20} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when user is below budget", () => {
    const tree = create(
      <SectionHeader co2value={10} monthlyCarbonBudget={20} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when user has budget below 1%", () => {
    const tree = create(
      <SectionHeader co2value={1} monthlyCarbonBudget={1000} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when co2value equals 0", () => {
    const tree = create(
      <SectionHeader co2value={0} monthlyCarbonBudget={1} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when monthlyCarbonBudget equals 0", () => {
    const tree = create(
      <SectionHeader co2value={1} monthlyCarbonBudget={0} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when monthlyCarbonBudget and co2value equals 0", () => {
    const tree = create(
      <SectionHeader co2value={0} monthlyCarbonBudget={0} date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
