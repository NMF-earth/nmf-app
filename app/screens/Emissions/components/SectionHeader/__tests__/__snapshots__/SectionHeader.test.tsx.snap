// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader should render when date 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#E7F3EB",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "transparent",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    blue={true}
    bold={true}
    style={
      {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <Ionicons
    color="#3957D0"
    name="ios-chevron-forward-outline"
    size={24}
    style={{}}
  />
</TouchableOpacity>
`;

exports[`SectionHeader should render when title 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#E7F3EB",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "transparent",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    blue={true}
    bold={true}
    style={
      {
        "textTransform": "capitalize",
      }
    }
  >
    some title
  </Text.Primary>
  <Ionicons
    color="#3957D0"
    name="ios-chevron-forward-outline"
    size={24}
    style={{}}
  />
</TouchableOpacity>
`;
